support/Pipelines/Bbh/ControlId.py

```diff
@@ -0,0 +1,173 @@
+    chi_target_A = binary_data["ObjectRight"]["KerrSchild"]["Spin"]
+    chi_target_B = binary_data["ObjectLeft"]["KerrSchild"]["Spin"]
+    x_B, x_A = binary_data["XCoords"]
+    separation = x_A - x_B
+    orbital_angular_velocity = binary_data["AngularVelocity"]
+    radial_expansion_velocity = binary_data["Expansion"]
+
+    # File to write control diagnostic data
+    data_file = open(f"{id_run_dir}/ControlParamsData.txt", "w")
+
+    iteration = 0
+    control_run_dir = id_run_dir
+
+    # Function to be minimized
+    def Residual(
+        M_Kerr_A: float,
+        M_Kerr_B: float,
+        chi_Kerr_A_x: float,
+        chi_Kerr_A_y: float,
+        chi_Kerr_A_z: float,
+        chi_Kerr_B_x: float,
+        chi_Kerr_B_y: float,
+        chi_Kerr_B_z: float,
+    ):
+        nonlocal iteration
+        nonlocal control_run_dir
+
+        if iteration > 0:
+            print(
+                "\n"
+                "=========================================="
+                f" Control of BBH Parameters ({iteration}) "
+                "=========================================="
+            )
+            control_run_dir = f"{id_run_dir}/ControlParams{iteration:02}"
+
+            # Run ID and find horizons
+            generate_id(
+                mass_a=M_Kerr_A,
+                mass_b=M_Kerr_B,
+                dimensionless_spin_a=[chi_Kerr_A_x, chi_Kerr_A_y, chi_Kerr_A_z],
+                dimensionless_spin_b=[chi_Kerr_B_x, chi_Kerr_B_y, chi_Kerr_B_z],
+                separation=separation,
+                orbital_angular_velocity=orbital_angular_velocity,
+                radial_expansion_velocity=radial_expansion_velocity,
+                run_dir=control_run_dir,
+                scheduler=None,
+            )
+
+        # Get black hole physical parameters
+        horizons_file = h5py.File(f"{control_run_dir}/Horizons.h5", "r")
+        M_A = horizons_file["AhA.dat"][0, 4]  # ChristodoulouMass
+        M_B = horizons_file["AhB.dat"][0, 4]  # ChristodoulouMass
+        chi_A_x = horizons_file["AhA.dat"][0, 6]  # DimensionlessSpinVector_x
+        chi_A_y = horizons_file["AhA.dat"][0, 7]  # DimensionlessSpinVector_y
+        chi_A_z = horizons_file["AhA.dat"][0, 8]  # DimensionlessSpinVector_z
+        chi_B_x = horizons_file["AhB.dat"][0, 6]  # DimensionlessSpinVector_x
+        chi_B_y = horizons_file["AhB.dat"][0, 7]  # DimensionlessSpinVector_y
+        chi_B_z = horizons_file["AhB.dat"][0, 8]  # DimensionlessSpinVector_z
+
+        residual = np.array(
+            [
+                M_A - M_target_A,
+                M_B - M_target_B,
+                chi_A_x - chi_target_A[0],
+                chi_A_y - chi_target_A[1],
+                chi_A_z - chi_target_A[2],
+                chi_B_x - chi_target_B[0],
+                chi_B_y - chi_target_B[1],
+                chi_B_z - chi_target_B[2],
+            ]
+        )
+        print("Control Residual =", residual)
```

Comment:
You can get nice formatting with `f"Control residual = {residual:e}"` (residuals are best displayed in exponential notation)